src/components/OrderBy.component.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import {
+  TiArrowSortedDown,
+  TiArrowSortedUp,
+  TiArrowUnsorted,
+} from "react-icons/ti";
+import { useSearchParams } from "react-router-dom";
+import styled from "styled-components";
+
+import { Order_By } from "../graphql/generated/graphql";
+import { defaultTheme } from "../theme";
+import { URL_PARAMS } from "../utils/constants";
+import { ButtonOnHoverOppacity, H4 } from "./styled";
+
+interface Props {
+  toggleOrder: () => void;
+}
+
+export const OrderBy = ({ toggleOrder }: Props) => {
+  const [search] = useSearchParams();
+  const activeSortInUrl = search.get(URL_PARAMS.SALE_DATE_SORT);
+
+  const currentSort =
+    activeSortInUrl === Order_By.Asc ? (
+      <TiArrowSortedUp />
+    ) : activeSortInUrl === Order_By.Desc ? (
+      <TiArrowSortedDown />
+    ) : (
+      <TiArrowUnsorted />
+    );
+
+  async function handleClick() {
+    toggleOrder();
+  }
+
+  return (
+    <H4>
+      Order By:
+      <OrderButton onClick={handleClick}>{currentSort}Sale Date</OrderButton>
```

Comment:
why don't we just call the toggleOrder here?

Reply:
fixed! I used to have more logic inside the handleClick, but now that it only does one thing it should be called the way you pointed